tests/test_endpoint.py

```diff
@@ -158,6 +158,13 @@ def checkEndpointBasics(self, endpoint, pointType, nAxes):
         pointDataRoundTrip = endpoint.dataFromPoint(point)
         assert_allclose(pointData, pointDataRoundTrip, err_msg=baseMsg)
 
+    def testEndpointEquals(self):
+        """Test Endpoint == Endpoint
+        """
+        for i1, point1 in enumerate(self.makeEndpoints()):
+            for i2, point2 in enumerate(self.makeEndpoints()):
+                self.assertEqual(i1 == i2, point1 == point2)
```

Comment:
Cute! But the message in case of an error will not be clear. I suggest formatting a useful error message to be passed as the `msg` argument to `self.assertEqual`. (This might be easier if you handled the `i1 == i2` case separately from the other case, but I'll leave that to you).

Also, please test `!=` as well as `==`.